### PR TITLE
Allow RGB with VIA + Fix CAPS lock led indicator

### PR DESCRIPTION
**Pull request by BestPig**
## Description

 - Add the ability to configure the RGB through VIA
 - Fix CAPS lock led status indicator

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/k1/rgb/se/config.h [hunk 1]
```diff
@@ -40,7 +40,7 @@
 #define DEBOUNCE 5
 
 /* LED Status indicators */
-#define LED_CAPS_LOCK_PIN D4
+#define LED_CAPS_LOCK_PIN B12
```
Comment (WillWill56) [inline keyboards/keychron/k1/rgb/se/config.h:43]: The caps lock indicator LED was working fine on my keyboard, seems like your keyboard uses a different pin? Probably need to move this into the mechanical/optical config files so it changes depending on the keyboard chosen. I can fix this after merging if you'd like?

**Review by WillWill56 — COMMENTED** [review]
Sorry for the delay, didn't see the notification for this request and been a bit absent in general. Some useful changes in here, thanks!

File: keyboards/keychron/k1/rgb/se/config.h (hunk 1 above)
Comment (WillWill56) [reply keyboards/keychron/k1/rgb/se/config.h:43]: I might as well do that now, seems like the best option.

Comment (BestPig) [reply keyboards/keychron/k1/rgb/se/config.h:43]: Oh I forgot your previous message. Yes I think it's the better option ;).
It's perfectly fine for me if you do it.

**Pull request merged**